ibis/backends/dask/execution/selection.py

```diff
@@ -0,0 +1,201 @@
+"""Dispatching code for Selection operations.
+"""
+
+from __future__ import absolute_import
+
+import functools
+import operator
+from typing import Optional
+
+import dask.dataframe as dd
+import numpy as np
+import pandas
+from toolz import concatv
+
+import ibis.expr.operations as ops
+import ibis.expr.types as ir
+from ibis.backends.pandas.execution.selection import (
+    _compute_predicates,
+    compute_projection,
+    compute_projection_table_expr,
+    execute,
+    execute_node,
+    map_new_column_names_to_data,
+    remap_overlapping_column_names,
+)
+from ibis.expr.scope import Scope
+from ibis.expr.typing import TimeContext
+
+from ..execution import constants
+
+
+@compute_projection.register(ir.ScalarExpr, ops.Selection, dd.DataFrame)
+def compute_projection_scalar_expr(
+    expr,
+    parent,
+    data,
+    scope: Scope = None,
+    timecontext: Optional[TimeContext] = None,
+    **kwargs,
+):
+    name = expr._name
+    assert name is not None, 'Scalar selection name is None'
+
+    op = expr.op()
+    parent_table_op = parent.table.op()
+
+    data_columns = frozenset(data.columns)
+
+    scope = scope.merge_scopes(
+        Scope(
+            {
+                t: map_new_column_names_to_data(
+                    remap_overlapping_column_names(
+                        parent_table_op, t, data_columns
+                    ),
+                    data,
+                )
+            },
+            timecontext,
+        )
+        for t in op.root_tables()
+    )
+    scalar = execute(expr, scope=scope, **kwargs)
+    result = pandas.Series([scalar], name=name).repeat(len(data.index))
+    result.index = data.index
+    return dd.from_pandas(result, npartitions=1)
+
+
+@compute_projection.register(ir.ColumnExpr, ops.Selection, dd.DataFrame)
+def compute_projection_column_expr(
+    expr,
+    parent,
+    data,
+    scope: Scope,
+    timecontext: Optional[TimeContext],
+    **kwargs,
+):
+    result_name = getattr(expr, '_name', None)
+    op = expr.op()
+    parent_table_op = parent.table.op()
+
+    if isinstance(op, ops.TableColumn):
+        # slightly faster path for simple column selection
+        name = op.name
+
+        if name in data:
+            return data[name].rename(result_name or name)
+
+        if not isinstance(parent_table_op, ops.Join):
+            raise KeyError(name)
+        (root_table,) = op.root_tables()
+        left_root, right_root = ops.distinct_roots(
+            parent_table_op.left, parent_table_op.right
```

Comment:
What is the type of `parent_table_op` here?

Reply:
Depends - at a high level it is an ibis.expr, though the exact op depends on the context with which we entered this function.